DMCompiler/DM/Expressions/Dereference.cs

```diff
@@ -85,6 +88,7 @@ public Dereference(Location location, DreamPath? path, DMExpression expression,
                         ShortCircuitHandler(proc, endLabel, shortCircuitMode);
                     }
                     proc.DereferenceField(fieldOperation.Identifier);
+                    _valType = dmObject.GetVariable(fieldOperation.Identifier)?.ValType ?? DMValueType.Anything;
```

Comment:
This tries to get the type from a var on the DMObject the proc is in rather than the thing actually being dereferenced